### ISSUE TITLE
Export createType as api.type(...)

### ISSUE THREAD

**Issue by jacogr** [polkadot-js/api#1213]
Mention `createType` as removed in the future

**Comment by amaurymartiny** [comment]
As discussed, I would prefer this signature:
`api.type(<Name>, <value>, <additionalOptions>)`

where `value` is always one arg, and `additionalOptions` are other optional stuff that user can pass into the type, along with metadata,typeregistry,networkId etc which are always passed

**Comment by amaurymartiny** [comment]
Just thinking: we also need to create older types (with 1328): either `api.type.at('hash')`, or `api.type('name', 'value', { at: 'hash' })`

**Comment by jacogr** [comment]
@amaurymartiny Yes, the additional options are needed, i.e. the raw is used in this form in api, apps & extension (for anywhere where the extrinsic version is required) -

https://github.com/polkadot-js/api/blob/master/packages/api/test/util/SingleAccountSigner.ts#L44

**Comment by axelchalon** [comment]
I like the idea! However `api.type` is already in use

```ts
/**
 * @description The type of this API instance, either 'rxjs' or 'promise'
 */
public get type (): ApiTypes {
  return this._type;
}
```

**Comment by jacogr** [comment]
Happy to break that, it is mostly only internal. Even something horrible like `api.apiType` works - and since it is a getter, won't have issues with `type(...)` the function.
